### ISSUE TITLE
x/tools: failures with `junk at end of line:` on linux-s390x-ibm

### ISSUE THREAD

**Issue by bcmills** [golang/go#52408]
```
package nosuchpkg is not in GOROOT (/data/golang/workdir/go/src/nosuchpkg)
package nosuchpkg is not in GOROOT (/data/golang/workdir/go/src/nosuchpkg)
open /data/golang/workdir/gopath/src/golang.org/x/tools/go/loader/missing.go: no such file or directory
open /data/golang/workdir/gopath/src/golang.org/x/tools/go/loader/missing.go: no such file or directory
/data/golang/workdir/gopath/src/golang.org/x/tools/go/loader/testdata/badpkgdecl.go:1:34: expected 'package', found 'EOF'
/data/golang/workdir/gopath/src/golang.org/x/tools/go/loader/testdata/badpkgdecl.go:1:34: expected 'package', found 'EOF'
/go/src/b/x.go:1:21: could not import c (cannot find package "c" in any of:
	/go/src/c (from $GOROOT)
	($GOPATH not set. For more details see: 'go help gopath'))
/go/src/b/x.go:1:21: could not import c (cannot find package "c" in any of:
	/go/src/c (from $GOROOT)
	($GOPATH not set. For more details see: 'go help gopath'))
/go/src/b/x.go:1:21: could not import c (/go/src/c/x.go:1:8: expected 'IDENT', found 'EOF')
/go/src/c/x.go:1:20: expected operand, found 'EOF'
cannot find package "two/three" in any of:
	/go/src/two/three (from $GOROOT)
	($GOPATH not set. For more details see: 'go help gopath')
cannot find package "http" in any of:
	/go/src/vendor/http (vendor tree)
	/go/src/http (from $GOROOT)
	($GOPATH not set. For more details see: 'go help gopath')
/go/src/c/x.go:1:31: cannot convert false (untyped bool constant) to int
/data/golang/workdir/tmp/ccgcZQ7Z.s: Assembler messages:
/data/golang/workdir/tmp/ccgcZQ7Z.s:26: Error: junk at end of line: `,3'
/data/golang/workdir/tmp/ccgcZQ7Z.s:27: Error: junk at end of line: `,3'
/data/golang/workdir/tmp/ccgcZQ7Z.s:63: Error: junk at end of line: `,3'
/data/golang/workdir/tmp/ccgcZQ7Z.s:64: Error: junk at end of line: `,3'
cgo failed: [go tool cgo -objdir /data/golang/workdir/tmp/golang.org_x_tools_go_loader_testdata_issue46877_C2592312503 -- -I /data/golang/workdir/tmp/golang.org_x_tools_go_loader_testdata_issue46877_C2592312503 x.go]: exit status 1
--- FAIL: TestCgoCwdIssue46877 (0.22s)
    loader_test.go:842: Load failed: couldn't load packages due to errors: golang.org/x/tools/go/loader/testdata/issue46877
FAIL
FAIL	golang.org/x/tools/go/loader	2.196s
```

This error message is also seen in several other tests, probably from the same root cause.

`greplogs --dashboard -md -l -e 'junk at end of line' --since=2022-01-01`

[2022-04-18T12:04:50-46bc274-91b9915/linux-s390x-ibm](https://build.golang.org/log/d8a6d4efae53d321c15bbf5e83ea3140cd424f15)
[2022-04-17T00:58:13-46bc274-e948c33/linux-s390x-ibm](https://build.golang.org/log/24ce3032f98d61d7d96e50655b9fd14aa5cc1846)
[2022-04-16T02:01:19-46bc274-5e4543c/linux-s390x-ibm](https://build.golang.org/log/1770233e52a5af429fc669821e8999c9cda7c6b2)
[2022-04-15T23:25:46-46bc274-e398266/linux-s390x-ibm](https://build.golang.org/log/12c40d3eaceaa60ee1fba001a8753fef7164d4f6)

**Comment by bcmills** [comment]
(attn @jonathan-albrecht-ibm)

**Comment by jonathan-albrecht-ibm** [comment]
So far these failures are only happening on one of the builders (go02). I can reproduce it manually on go02 as well.

It's not happening on the other builder (go01) either as part of the build farm builds or when I build and test manually.

I've stopped the go02 builder service while I try and figure out what's going wrong.

**Comment by jonathan-albrecht-ibm** [comment]
The problem with the go02 build seems to be because some of the gcc* packages were updated due to: https://go-review.googlesource.com/c/build/+/399996. I think it left the gcc toolchain in a bad state. I rolled back the updated packages and tested locally and everything worked again.

I've restarted the buildlet service on go02 and I'll watch it for problems. I've left a comment on https://go-review.googlesource.com/c/go/+/336549/comments/eb61ef44_7b74dba0 and I'll try and get things set up properly on the current builders. I'll try and set up the new builders I'm provisioning to avoid this mess.
